app/src/androidTest/java/ch/epfl/skysync/viewmodel/FlightsViewModelTest.kt

```diff
@@ -0,0 +1,212 @@
+    composeTestRule.setContent {
+      viewModel =
+          FlightsViewModel.createViewModel(
+              flightTable, balloonTable, basketTable, flightTypeTable, vehicleTable)
+
+      // val flights = viewModel.currentFlights.collectAsStateWithLifecycle()
+      // Text(flights.value.getOrNull(0)?.toString()?: "No flight types")
+    }
+  }
+
+  @Test
+  fun fetchesCurrentFlightsIfEmpty() {
+    val currentFlights = viewModel.currentFlights.value
+    Assert.assertEquals(currentFlights.size, 0)
+  }
+
+  @Test
+  fun fetchesCurrentFlights() {
+    var flight1 =
+        PlannedFlight(
+            nPassengers = 2,
+            team =
+                Team(
+                    roles =
+                        listOf(
+                            Role(RoleType.PILOT, dbSetup.pilot1),
+                            Role(RoleType.CREW, dbSetup.crew1))),
+            flightType = dbSetup.flightType2,
+            balloon = dbSetup.balloon1,
+            basket = dbSetup.basket2,
+            date = LocalDate.of(2024, 8, 12),
+            timeSlot = TimeSlot.AM,
+            vehicles = listOf(dbSetup.vehicle1),
+            id = UNSET_ID)
+
+    var flight2 =
+        PlannedFlight(
+            nPassengers = 2,
+            team =
+                Team(
+                    roles =
+                        listOf(
+                            Role(RoleType.PILOT, dbSetup.pilot1),
+                            Role(RoleType.CREW, dbSetup.crew1))),
+            flightType = dbSetup.flightType1,
+            balloon = dbSetup.balloon1,
+            basket = dbSetup.basket2,
+            date = LocalDate.of(2024, 8, 12),
+            timeSlot = TimeSlot.AM,
+            vehicles = listOf(dbSetup.vehicle1),
+            id = UNSET_ID)
+
+    flightTable.add(flight1, { flight1 = flight1.copy(it) }, { error -> println(error) })
+
+    flightTable.add(flight2, { flight2 = flight2.copy(it) }, { error -> println(error) })
+
+    SystemClock.sleep(DB_SLEEP_TIME)
+
+    viewModel.refreshCurrentFlights()
+    SystemClock.sleep(DB_SLEEP_TIME)
+    Assert.assertEquals(viewModel.currentFlights.value.size, 2)
+  }
+
+  @Test
+  fun addsFlight() {
+    var flight1 =
+        PlannedFlight(
+            nPassengers = 2,
+            team =
+                Team(
+                    roles =
+                        listOf(
+                            Role(RoleType.PILOT, dbSetup.pilot1),
+                            Role(RoleType.CREW, dbSetup.crew1))),
+            flightType = dbSetup.flightType2,
+            balloon = dbSetup.balloon1,
+            basket = dbSetup.basket2,
+            date = LocalDate.of(2024, 8, 12),
+            timeSlot = TimeSlot.AM,
+            vehicles = listOf(dbSetup.vehicle1),
+            id = UNSET_ID)
+
+    viewModel.addFlight(flight1)
+    SystemClock.sleep(DB_SLEEP_TIME)
+
+    viewModel.refreshCurrentFlights()
+    SystemClock.sleep(DB_SLEEP_TIME)
+    Assert.assertEquals(viewModel.currentFlights.value.size, 1)
```

Comment:
The DB is note reset before each test. Hence this test fails as there are 3 flights (2 from previous test and the new added)

Reply:
there is a clearDatabse() in the @Begin of the test function